ccmlib/common.py

```diff
@@ -76,6 +77,36 @@ class UnavailableSocketError(CCMError):
     pass
 
 
+class LogPatternToVersion(object):
+    def __init__(self, patterns_to_versions):
+        assert type(patterns_to_versions) is dict, "LogPatternToVersion takes a dictionary."
+        self.versions_to_patterns = patterns_to_versions
+
+    def __call__(self, version):
+        pattern = None
+        for key in self.versions_to_patterns.keys():
+                if version >= key:
+                    pattern = self.versions_to_patterns[key]
+        return pattern
+
+    def __iter__(self):
+        return iter(self.versions_to_patterns)
+
+    def __len__(self):
+        return len(self.versions_to_patterns)
+
+    def __repr__(self):
+        return str(self.versions_to_patterns)
+
+    @property
+    def patterns(self):
+        return [self.versions_to_patterns[version] for version in self.__iter__()]
```

Comment:
I think this could be simplified to `return list(self.version_to_patterns.values())`[